### ISSUE TITLE
Stylable & Storybook: Cannot read property 'root' of undefined

### ISSUE THREAD

**Issue by timvonwerne** [wix/stylable#1698]
I'm trying to implement storybook in a Stylable project and have created a button component to try it out:
```ts
import { st, classes } from './Button.st.css'
...
export class Button extends PureComponent<ButtonProps> {
  ...
  render() {
    ...
    return (
      <button type="button" className={st(classes.root, {priority, size})}>{children}</button>
    )
  }
}
```
```
**Cannot read property 'root' of undefined**
TypeError: Cannot read property 'root' of undefined
    at Button.render (http://localhost:6006/main.10c3d77cc995a0fad71f.bundle.js:522:135)
    at finishClassComponent (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:129354:31)
    at updateClassComponent (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:129304:24)
    at beginWork (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:130942:16)
    at HTMLUnknownElement.callCallback (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:115814:14)
    at Object.invokeGuardedCallbackDev (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:115863:16)
    at invokeGuardedCallback (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:115925:31)
    at beginWork$1 (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:135828:7)
    at performUnitOfWork (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:134640:12)
    at workLoopSync (http://localhost:6006/vendors~main.10c3d77cc995a0fad71f.bundle.js:134571:5)
```

When creating the story, the render function of the button component does not seem to know the classes object? Is there something I'm missing? Do I have to change the webpack configuration of Storybook to make this work?

**Comment by barak007** [comment]
Storybook doesn't support Stylable out of the box. You should add our plugin to the storybook webpack config.

**Comment by timvonwerne** [comment]
Cheers for your quick response @barak007. So I just need to add the StylableWebpackPlugin to Storybook's main.js like this?

```js
const { StylableWebpackPlugin } = require('@stylable/webpack-plugin');

module.exports = {
  "stories": [
    "../src/**/*.stories.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials"
  ],
  "core": {
    "builder": "webpack5"
  },
  webpackFinal: (config) => {
    config.plugins.push(new StylableWebpackPlugin());
    return config;
  },
}
```

Because it seems like Storybook is trying to use postcss to work with the *.st.css files and fails with it..

**Comment by barak007** [comment]
I see. In order for this to work all css loaders should ignore .st.css files. I will try to configure storybook and will back with the response.

**Comment by barak007** [comment]
I have few questions:
1. What version of storybook do you use?
2. What version of Stylable do you use?
3. Can you provide the project that has the error for inspection?

10x in advance.

**Comment by timvonwerne** [comment]
It seems like there are a couple of misunderstandings I have here as I'm totally new to Stylable, Webpack & Storybook..
My project uses Webpack 5 which Storybook is not (yet) compatible with. When I use Webpack 4 I get the error that makeSerializable is not found. When using Webpack 5 and the just released Storybook beta I get the error which is described in the GitHub repo I'll link below.

Now to your questions:
1. Storybook version is 6.2.0-beta.13
2. Stylable version is 4.0.1

As I said the Webpack version I'm using is 5.24.4.

Here's the link to a test repo: [https://github.com/timvonwerne/storybook-problem](https://github.com/timvonwerne/storybook-problem)
I understand that the problems I have are not really related to Stylable. Nevertheless I am grateful for any help.

**Comment by barak007** [comment]
Ok some news. I think I found the root cause. I will open PR with the fix soon. (probably early next week)

**Comment by tomrav** [comment]
We've just released version `v4.0.2` across the Stylable stack, among other things it includes a fix for the storybook setup.

You can read more about integrating Stylable to Storybook here [here](https://stylable.io/docs/getting-started/storybook-integration).

**Comment by timvonwerne** [comment]
Thank you guys for your work, @tomrav @barak007! I just tested it out in my demo repository and can confirm it works perfectly fine 👍🏻
Not yet in my actual project, but I'm confident that I'll manage that too.
